File: backend/app/schemas/symptom.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


SymptomCategory = Literal[
    "cramps",
    "headache",
    "mood",
    "fatigue",
    "sleep",
    "bloating",
    "nausea",
    "other",
]


class SymptomCreate(BaseModel):
    date_time: datetime
    category: SymptomCategory
    severity: int = Field(ge=1, le=10)
    notes: str | None = None
    tags: list[str] | None = None


class SymptomUpdate(BaseModel):
    date_time: datetime | None = None
    category: SymptomCategory | None = None
    severity: int | None = Field(default=None, ge=1, le=10)
    notes: str | None = None
    tags: list[str] | None = None


class SymptomResponse(BaseModel):
    id: str
    user_id: str
    date_time: datetime
    category: str
    severity: int
    notes: str | None
    tags: list[str] | None
    created_at: datetime

File: frontend/src/types/symptom.ts
export type SymptomCategory =
  | "cramps"
  | "headache"
  | "mood"
  | "fatigue"
  | "sleep"
  | "bloating"
  | "nausea"
  | "other";

export type Symptom = {
  id: string;
  user_id: string;
  date_time: string;
  category: SymptomCategory;
  severity: number;
  notes: string | null;
  tags: string[] | null;
  created_at: string;
};

export type CreateSymptomRequest = {
  date_time: string;
  category: SymptomCategory;
  severity: number;
  notes?: string;
  tags?: string[];
};

export type UpdateSymptomRequest = {
  date_time?: string;
  category?: SymptomCategory;
  severity?: number;
  notes?: string;
  tags?: string[];
};